### DISCUSSION TITLE
Add a new setValidationSchema API

### DISCUSSION THREAD

**Discussion by bombillazo** [react-hook-form/react-hook-form#8783] [Ideas] [answered]
It would be great if we could change the validation schema programmatically after the `useForm` hook has been called. I was thinking it can be a new option returned by `useform`/`useFormContext`.

**Comment by bluebill1049** [comment, marked as answer]
hey @bombillazo have you tried building a `resolver` with `context` switching?

eg: https://codesandbox.io/s/react-hook-form-customresoliver-v7-8mpqw

**Reply by bombillazo** [reply]
Yeah right now I'm using another context to set the validation schema, but it would've been nice to do so directly from the `useFormContext` hook.

**Reply by bombillazo** [reply]
I guess I'll mark this as the answer, even though the idea still stands to have an API to directly change the validation schema using the hook.
